src/transactionsFactories/transferTransactionsFactory.ts

```diff
@@ -17,23 +36,66 @@ interface IConfig {
     gasLimitMultiESDTNFTTransfer: bigint;
 }
 
-interface TokenComputer {
+interface ITokenComputer {
     isFungible(token: Token): boolean;
 }
 
+interface IGasEstimator {
+    forEGLDTransfer(dataLength: number): number;
+    forESDTTransfer(dataLength: number): number;
+    forESDTNFTTransfer(dataLength: number): number;
+    forMultiESDTNFTTransfer(dataLength: number, numTransfers: number): number;
+}
+
 /**
  * Use this class to create transactions for native token transfers (EGLD) or custom tokens transfers (ESDT/NTF/MetaESDT).
  * This name is only temporary, the class will be renamed to `TransferTransactionsFactory`.
  */
-export class NextTransferTransactionsFactory {
-    private readonly config: IConfig;
-    private readonly dataArgsBuilder: TokenTransfersDataBuilder;
-    private readonly tokenComputer: TokenComputer;
-
-    constructor(options: { config: IConfig; tokenComputer: TokenComputer }) {
-        this.config = options.config;
-        this.tokenComputer = options.tokenComputer;
-        this.dataArgsBuilder = new TokenTransfersDataBuilder();
+export class TransferTransactionsFactory {
+    private readonly config?: IConfig;
+    private readonly dataArgsBuilder?: TokenTransfersDataBuilder;
+    private readonly tokenComputer?: ITokenComputer;
+    private readonly gasEstimator?: IGasEstimator;
+
+    constructor(options: IGasEstimator | { config: IConfig; tokenComputer: ITokenComputer }) {
```

Comment:
Can add extra comment (documentation comment) to mention that the variant with gas estimator is legacy.

Reply:
Done